V almonacid/ch2285/add support for legacy utxo tx

v-almonacid opened the pull request:
This PR contains functions that are not yet implemented in the rust bindings and also a few other things have been recently modified in `js-chain-libs`. I open this PR anyway to help keeping track of the changes and ease comparison with the history of additions by seba in front-end.

v-almonacid committed:
add missing test and fix addWitnesses

v-almonacid committed:
add support for legacy utxo tx

v-almonacid committed:
flow

v-almonacid committed:
fix and clean up

nicarq commented:
@v-almonacid you removed the `[WIP / DO NOT MERGE]` label, do that means that this could be merged?

v-almonacid commented:
@nicarq yes all PR's should be merged in the right order (even if they have deprecated/non functional code), I'm adding the changes more or less in the same way as in frontend. First I added the WIP label because I thought I would do everything in the same PR, but then I changed my mind.
